src/pds/ingress/pds-nucleus-product-completion-checker.py

```diff
@@ -0,0 +1,321 @@
+                save_incomplete_product(s3_url_of_product_label, missing_files)
+            else:
+                product_completed(s3_url_of_product_label)
+                
+        # Data file received
+        else:
+            
+            if not s3_url_of_data_file.lower().endswith("/"): # Not a directory
+                copy_file_to_efs(bucket, key, efs_mount_path)
+                save_received_file(s3_url_of_data_file)
+                update_existing_incomplete_products(s3_url_of_data_file)
+                delete_expected_file(s3_url_of_data_file)
+            
+        return 'Copied'
+    except Exception as e:
+        print(e)
+        # print('Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket))
+        raise e
+              
+
+# Handles the missing files detected for a product label
+def handle_missing_files_for_product_label(s3_url_of_product_label, bucket, key):
+    s3_base_dir = s3_url_of_product_label.rsplit('/',1)[0]
+    
+    print('Get s3_response for key {} from bucket {}'.format(key, bucket))
+    
+    try:
+        s3_response = s3.get_object(Bucket=bucket, Key=key)
+    
+        # Get the Body object in the S3 get_object() response
+        s3_object_body = s3_response.get('Body')
+    
+        # Read the data in bytes format and convert it to string
+        content_str = s3_object_body.read().decode()
+    
+        # parse xml
+        xmldoc = minidom.parseString(content_str)
+        missing_files_from_product_label = xmldoc.getElementsByTagName('file_name')
+        
+        missing_files = []
+        
+        for x in missing_files_from_product_label:
+
+            s3_url_of_data_file = s3_base_dir + "/" + x.firstChild.nodeValue
+    
+            # Check received file table
+            items = check_in_received_file_table(s3_url_of_data_file)
+    
+            # File is already received
+            if items:
+                print('The file is already received: ' + items['s3_url_of_data_file'])
+            else:
+    
+                missing_files.append(str(s3_url_of_data_file))
+                update_expected_files(s3_url_of_data_file, s3_url_of_product_label)
+                print("Missing files: " + str(missing_files))
+
+        return missing_files
+
+    except Exception as e:
+        print(e)
+        # print('Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket))
+        raise e
+
+
+# Saves received files
+def save_received_file(s3_url_of_data_file):
+    received_data_files_table = dynamodb.Table('received_data_files')
+
+    received_data_files_table.put_item(
+        Item={
+            's3_url_of_data_file': s3_url_of_data_file
+        }
+    )
+
+
+# Returns an expected files response (with list of product labels) for a given s3_url_of_data_file
+def get_expected_file_items(s3_url_of_data_file):
```

Comment:
We should always get a single item from this function, I think that would simplify the understanding of the code if we make the function return a single `expected_file`.

Reply:
Or actually it sounds like it returns a single item, so maybe you only need to rename the function `get_expected_file`.

Reply:
@tloubrieu-jpl , changing it to `get_expected_file` since the code comment already explains the response

Reply:
Updated